generator/api/src/main/java/io/fabric8/maven/generator/api/MavenGeneratorContext.java

```diff
@@ -28,15 +31,18 @@
  */
 public class MavenGeneratorContext {
     private final MavenProject project;
+    private final MavenSession session;
+    private final GoalFinder goalFinder;
     private final ProcessorConfig config;
-
     private final Logger log;
     private final PlatformMode mode;
     private final OpenShiftBuildStrategy strategy;
 
-    public MavenGeneratorContext(MavenProject project, ProcessorConfig generatorConfig, Logger log,
+    public MavenGeneratorContext(MavenProject project, MavenSession session, GoalFinder goalFinder, ProcessorConfig generatorConfig, Logger log,
```

Comment:
Couldn't we feed in a `BuildMode` (like `watch` or `dist`) directly into the context or do the actual look in an `init()` step, exposing only this mode. So that the generator itself doesn't need todo this. Its also more the task of the surrounding execution environment to provide this information then for letting the generator doing the lookup.  Also the generator is then not coupled to specific goal names.